crypto/s2n_aead_cipher_chacha20_poly1305.c

```diff
@@ -157,6 +148,106 @@ static int s2n_aead_chacha20_poly1305_destroy_key(struct s2n_session_key *key)
     return 0;
 }
 
+#elif defined(S2N_CHACHA20_POLY1305_AVAILABLE_BSSL_AWSLC) /* BoringSSL and AWS-LC implementation */
+
+static int s2n_aead_chacha20_poly1305_encrypt(struct s2n_session_key *key, struct s2n_blob *iv, struct s2n_blob *aad, struct s2n_blob *in, struct s2n_blob *out)
+{
+    gte_check(in->size, S2N_TLS_CHACHA20_POLY1305_TAG_LEN);
+    gte_check(out->size, in->size);
+    eq_check(iv->size, S2N_TLS_CHACHA20_POLY1305_IV_LEN);
+
+    /* Adjust input length to account for the Tag length */
+    size_t in_len = in->size - S2N_TLS_CHACHA20_POLY1305_TAG_LEN;
+    size_t out_len = 0;
+
+    GUARD_OSSL(EVP_AEAD_CTX_seal(key->evp_aead_ctx, out->data, &out_len, out->size, iv->data, iv->size, in->data, in_len, aad->data, aad->size), S2N_ERR_ENCRYPT);
+
+    S2N_ERROR_IF((in_len + S2N_TLS_CHACHA20_POLY1305_TAG_LEN) != out_len, S2N_ERR_ENCRYPT);
```

Comment:
Should this be checked up in the OpenSSL version?

Reply:
What does @ttjsu-aws say?
There were no such checks originally.

Reply:
We could benefit from adding a check here.

Reply:
We could benefit from adding a check here for OpenSSL version.

Reply:
I have added similar checks for the OpenSSL implementation.